app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -226,4 +227,20 @@ class PatientsScreenController @Inject constructor(
         .filter { it.permissionResult == RuntimePermissionResult.GRANTED }
         .map { Ui::openScanSimpleIdCardScreen }
   }
+
+  private fun handleVisibilityOfSyncIndicator(events: Observable<UiEvent>): Observable<UiChange> {
+    val screenCreated = events.ofType<ScreenCreated>()
+
+    val canUserSync = { userSession.canSyncData().distinctUntilChanged() }
+
+    val userApprovedForSync = Observables.combineLatest(screenCreated, canUserSync())
+        .filter { (_, canUserSync) -> canUserSync }
+        .map { { ui: Ui -> ui.showSyncIndicator() } }
+
+    val userWaitingForApproval = Observables.combineLatest(screenCreated, canUserSync())
```

Comment:
`canUserSync()` gets called immediately when this function is called so the lambda can be removed. By doing so we can also share the stream between `userApprovedForSync` and `userWaitingForApproval`.

Reply:
Right!